src/test/java/org/elasticsearch/test/ElasticsearchIntegrationTest.java

```diff
@@ -1953,15 +1936,27 @@ protected String routingKeyForShard(String index, String type, int shard) {
      * Return settings that could be used to start a node that has the given zipped home directory.
      */
     protected Settings prepareBackwardsDataDir(Path backwardsIndex, Object... settings) throws IOException {
-        Path indexDir = newTempDirPath();
+        Path indexDir = createTempDir();
         Path dataDir = indexDir.resolve("data");
         try (InputStream stream = Files.newInputStream(backwardsIndex)) {
             TestUtil.unzip(stream, indexDir);
         }
         assertTrue(Files.exists(dataDir));
-        Path[] list = FileSystemUtils.files(dataDir);
+
+        // list clusters in the datapath, ignoring anything from extrasfs
+        final Path[] list;
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(dataDir)) {
+            List<Path> dirs = new ArrayList<>();
+            for (Path p : stream) {
+                if (!p.getFileName().toString().startsWith("extra")) {
```

Comment:
right, users don't have predefined patterns about what files are on their machine. that is the point of this. sadly, in this one place, since its test *infra itself* this is cleanest way to do it. 

but concrete tests should really not be looking at any patterns like this, or it defeats the purpose.